### PR TITLE
[ENH] forecasting metrics: ensure uniform support and testing for `sample_weight` parameter

### LINKED ISSUE #6412: [BUG] MeanAbsoluteError does not leverage horizon_weight
**Describe the bug**
The performance metric MeanAbsoluteError does not take into account horizon_weight when calling the class instance.
All other scikit-learn based metrics do.

**To Reproduce**

```python
import numpy as np
from sktime.performance_metrics.forecasting import MeanAbsoluteError
y_true = np.array([3, -0.5, 2, 7, 2])
y_pred = np.array([2.5, 0.0, 2, 8, 1.25])
mae = MeanAbsoluteError()
mae(y_true, y_pred) != mae(y_true, y_pred, horizon_weight=np.array([0.1,0.2,0.1,0.3,0.4]))
```

**Expected behavior**
The boolean statement at the end should be true.

**Additional context**

**Versions**
0.28.0

### PR DESCRIPTION
**Pull request by fkiraly**
Fixes #6412.

This adds a `sample_weight` argument to all `sktime` metrics, and restores the ability to pass `horizon_weight` for downwards compatibility.

For compatibility with `sklearn`, the publicly documented argument is called `sample_weight`.

### CODE REVIEW

**Comment by fkiraly** [comment]
@sktime/core-developers, error that fixes this has made it on `main` with last minor release and breaks downwards compatibility with 0.28.X and prior, so I would merge this before release to restore compatibility of code
